security/msal/MsalClient.ts

```diff
@@ -247,6 +267,82 @@ export class MsalClient extends BaseOAuthClient<MsalClientConfig, MsalTokenSpec>
             : await client.logoutPopup(opts);
     }
 
+    //------------------------
+    // Telemetry
+    //------------------------
+    enableTelemetry(): void {
+        if (this._telemetryCbHandle) {
+            this.logInfo('Telemetry already enabled', this.telemetryResults);
+            return;
+        }
+
+        this.telemetryResults = {
+            startTime: new Date(),
+            endTime: null,
+            events: {}
+        };
+
+        this._telemetryCbHandle = this.client.addPerformanceCallback(events => {
+            events.forEach(e => {
+                try {
+                    const {events} = this.telemetryResults,
+                        {name, startTimeMs, durationMs, success, errorName, errorCode} = e,
+                        eTime = startTimeMs ? new Date(startTimeMs) : new Date();
+
+                    if (!events[name]) {
+                        events[name] = {
+                            firstTime: eTime,
+                            lastTime: eTime,
+                            successCount: 0,
+                            failureCount: 0,
+                            duration: {count: 0, total: 0, average: 0, worst: 0},
+                            lastFailure: null
+                        };
+                    }
+
+                    const eResult = events[name];
+                    eResult.lastTime = eTime;
+
+                    if (success) {
+                        eResult.successCount++;
+                    } else {
+                        eResult.failureCount++;
+                        eResult.lastFailure = {
+                            time: eTime,
+                            code: errorCode,
+                            name: errorName,
+                            raw: e
+                        };
+                    }
+
+                    const {duration} = eResult;
+                    if (durationMs) {
+                        duration.count++;
+                        duration.total += durationMs;
+                        duration.average = duration.total / duration.count;
+                        duration.worst = Math.max(duration.worst, durationMs);
+                    }
+                } catch (e) {
+                    this.logError(`Error processing telemetry event`, e);
+                }
+            });
+        });
+
+        this.logInfo('Telemetry enabled');
+    }
+
+    disableTelemetry(): void {
```

Comment:
Do we envision ever doing this?  Don't feel strongly about it, am curious if its worth it